### PR TITLE
[spark] Add np_util

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

Add np_util

### CODE REVIEW

File: docker/spark/Dockerfile [hunk 1]
```diff
@@ -23,15 +23,20 @@ ARG PROTOBUF_VERSION=3.21.9
 COPY extensions/spark/setup/dist/ dist/
 RUN pip3 install --no-cache-dir dist/djl_spark-${DJL_VERSION}-py3-none-any.whl && \
     rm -rf dist
+RUN pip3 install --no-cache-dir pillow
```
Comment (lanking520) [inline docker/spark/Dockerfile:26]: You can combine all RUN command together. Each `RUN` statement written in Dockerfile means each layer addtion. We would like to have as less layer as possible

File: extensions/spark/setup/djl_spark/util/np_util.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
+# BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+import io
+import numpy as np
+
+
+def to_npz(arr):
```
Comment (frankfliu) [inline extensions/spark/setup/djl_spark/util/np_util.py:18]: We should support multiple array in npz file

File: extensions/spark/setup/djl_spark/util/np_util.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    return buf.read(-1)
+
+
+def from_npz(data):
+    """
+    Converts bytes to numpy array.
+    :param data: The bytes to convert.
+    :return: The result numpy array.
+    """
+    return np.load(io.BytesIO(data))['arr_0']
```
Comment (frankfliu) [inline extensions/spark/setup/djl_spark/util/np_util.py:36]: the name may not always be 'arr_0', we should all arrays in the zip file

File: docker/spark/Dockerfile (hunk 1 above)
Comment (xyang16) [reply docker/spark/Dockerfile]: Changed

File: extensions/spark/setup/djl_spark/util/np_util.py (hunk 2 above)
Comment (xyang16) [reply extensions/spark/setup/djl_spark/util/np_util.py]: Changed

**Review by frankfliu — APPROVED** [review]

**Pull request merged**